test/test_backend/db/tst_refresh_dataset_group_tables.py

```diff
@@ -0,0 +1,116 @@
+        n_lines_for_inputs = 50
+
+        for group, tables in cls.tables_by_group.items():
+            input_dir = THIS_INPUT_DIR / group
+            if not os.path.exists(input_dir):
+                os.makedirs(input_dir)
+
+            missing_inputs = set([f'{x}.csv' for x in tables]).difference(set(os.listdir(input_dir)))
+            if missing_inputs:
+                # Download dataasets if necessary
+                missing_downloads = set([f'{x}.csv' for x in tables]).difference(set(os.listdir(DOWNLOAD_DIR)))
+                if missing_downloads:
+                    refresh_dataset_group_tables(
+                        [group], download_only=True, schema=TEST_SCHEMA)
+                # Create inputs
+                for table in tables:
+                    with open(DOWNLOAD_DIR / f'{table}.csv', 'r') as f:
+                        sample_lines = f.readlines()[:n_lines_for_inputs]
+                    with open(input_dir / f'{table}.csv', 'w') as f:
+                        f.writelines(sample_lines)
+
+    def _analyze_group_tables(self, group_name: str, schema: str = TEST_SCHEMA) -> Dict:
+        """Analyze tables for a given dataset group
+        todo: this smells like it should maybe be refactored into a pure function"""
+        tables: List[str] = self.tables_by_group[group_name]
+        table_data = {}
+        index_statements_all: List[str] = get_ddl_statements(schema, ['indexes'], return_type='flat')
+        with get_db_connection(schema=schema) as con:
+            for table in tables:
+                indexes_expected: List[str] = [x for x in index_statements_all if f'{schema}.{table}(' in x]
+                indexes_and_pkeys: List[RowMapping] = sql_query(con, f"""
+                    SELECT indexname, indexdef
+                    FROM pg_indexes
+                    WHERE tablename = '{table}' AND schemaname = '{schema}';""")
+                pkeys: List[RowMapping] = [x for x in indexes_and_pkeys if 'pkey' in x['indexname']]
+                table_data[table] = {
+                    'indexes_expected': indexes_expected,
+                    'indexes_and_pkeys': indexes_and_pkeys,
+                    'pkeys': pkeys
+                }
+        return table_data
+
+    def _check_indexes_and_pkeys(self, group_name: str, schema: str = TEST_SCHEMA):
+        """Run tests on given dataset group and schema"""
+        table_data: Dict = self._analyze_group_tables(group_name, schema)
+        for table, data in table_data.items():
+            # - test: Expected indexes on each table
+            self.assertEqual(len(data['indexes_expected']), len(data['indexes_and_pkeys']) - len(data['pkeys']))
+            # - test: Concept table only: Has primary key(s)
+            if table == 'concept':
+                self.assertGreater(len(data['pkeys']), 0)
+
+    def test_current_vocab(self):
+        """Test current tables setup for group (indexes and, if applicable, primary keys): vocab"""
```

Comment:
## Test for running in action
Added tests not only to check that loading works correctly, but also to see if indexes/pkeys are properly set on the current vocab/counts tables. This runs at the end of actions.